files.go

```diff
@@ -196,6 +206,37 @@ func (api *Client) GetFiles(params GetFilesParameters) ([]File, *Paging, error)
 	return api.GetFilesContext(context.Background(), params)
 }
 
+// ListFiles retrieves all files according to the parameters given. Uses cursor based pagination.
+func (api *Client) ListFiles(params ListFilesParameters) ([]File, *ResponseMetadata, error) {
```

Comment:
return a single type here with a Next field that is ListFilesParameters.

Reply:
That's seems different than what other functions with cursor based paging already have implemented in the package. For example, the [GetUsersInConversationContext](https://godoc.org/github.com/nlopes/slack#Client.GetUsersInConversationContext) returns the cursor as a string as the second return argument.

Based on that finding, should it be this instead?

```
func (api *Client) ListFiles(params ListFilesParameters) ([]File, string, error)
```

Reply:
@james-lawrence any thoughts on ^^?

Reply:
sorry for the delays, I prefer the approach taken by [GetUsersPaginated](https://godoc.org/github.com/nlopes/slack#Client.GetUsersPaginated)

Reply:
implemented `return a single type here with a Next field that is ListFilesParameters.`, however left the name as `Cursor` to more closely align with the API parameter names.